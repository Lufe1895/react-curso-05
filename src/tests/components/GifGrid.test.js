import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Pruebas en el <GifGrid />', () => {
    const category = 'One Punch';
    test('Debe coincidir con el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();     
    });
    
    test('Debe mostrar items cuando se cargan imÃ¡genes', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://facebook.com/img.jpg',
            title: 'img'
        },
        {
            id: 'DEF',
            url: 'https://facebook.com/img.jpg',
            title: 'img2'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(true);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
});
