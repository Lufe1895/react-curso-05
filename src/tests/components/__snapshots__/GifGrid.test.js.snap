// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> Debe coincidir con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> Debe mostrar items cuando se cargan im√°genes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="img"
      url="https://facebook.com/img.jpg"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="img2"
      url="https://facebook.com/img.jpg"
    />
  </div>
</Fragment>
`;
